app/api/courses.js

```diff
@@ -0,0 +1,119 @@
+const { Router } = require('express')
+const course = require('../models/course')
+const router = Router()
+const db = require('../models/index')
+const { logger } = require('../../lib/logger')
+const courseService = require('../services/course_service')
+const enrollmentService = require('../services/enrollment_service')
+const sectionService = require('../services/section_service')
+const { generateUserAuthToken, requireAuthentication } = require('../../lib/auth')
+const e = require('express')
+
+router.use('/', require('./sections'))
+
+//GET request from /courses homepage
+router.get('/', requireAuthentication, async function (req, res) {
+    // TODO: use the authentication token (bearer) to authenticate & find the user
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const teacherCourses = await db.Course.findAll({
+        include: [{
+            model: db.Enrollment,
+            where: { role: 'teacher', userId: user.id }
+        }]
+    })
+    const studentCourses = await db.Course.findAll({
+        include: [
+            {
+                model: db.Section,
+                include: [
+                    {
+                        model: db.Enrollment,
+                        where: { role: 'student', userId: user.id },
+                    }
+                ]
+            }
+        ]
+    })
+
+    res.status(200).send({
+        studentCourses : courseService.extractArrayCourseFields(studentCourses),
+        teacherCourses : courseService.extractArrayCourseFields(teacherCourses)
+    })
+
+})
+
+//User Creates a course
+//Authenticate token, create course & create enrollment for logged in user as teacher in the course
+router.post('/', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+
+    // create course & role as a teacher in enrollment
+    // create course 
+    if (req.body.name) { // name is required, other fields can be left unspecified
+        // create course with valid fields
+        const course = await db.Course.create(courseService.extractCourseFields(req.body))
+        
+        //create the enrollment
+        const enrollmentToInsert = {
+            courseId: course.id,
+            userId: user.id,
+            role: 'teacher'
+            // no section because they are a teacher
+        }
+        const missingFields = enrollmentService.validateEnrollmentCreationRequest(enrollmentToInsert)
+        if (missingFields.length == 0) {
+            const enrollment = await db.Enrollment.create(enrollmentToInsert)
+            res.status(201).send({
```

Comment:
should we add a check to ensure the user is a teacher here?

Reply:
I don't think so. A user can be a teacher and a student at the same time in different courses. The whole ideas is that the teacher of a course is just whoever made the course. Does that make sense? That's why the role ("teacher" or "student") is stored in the enrollment table rather than the user table